test/libsolidity/semanticTests/builtins/balance_with_balance.sol

```diff
@@ -0,0 +1,8 @@
+contract ClientReceipt {
+    constructor() payable {}
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor(), 1000 wei ->
+// contract.balance -> 1000
```

Comment:
Hmm, can we make this symmetric an include the denomination in the result, i.e.` -> 1000 wei`?

Reply:
True, I will add support for this. I think it should not be too difficult.

Reply:
I think it is better to add support for the denominators in another PR.

Reply:
Why?

Reply:
I think denominations are supported on the input side already, and on the output side it is not possible to tell what it should be, but I think the printer should always use the `wei` suffix. Test writes can choose to fill it out with ether (or other denom) which should be parsed properly and be accepted.

Reply:
Oh I see, basically you'd need to store the ABI in the builtin list for the test format parser to handle it?

Reply:
Not sure, at least it didn't looked like a super simple fix. I would prefer to merge this first, and improve later.